apps/deploy_monitoring/src/monitors/mod.rs

```diff
@@ -0,0 +1,146 @@
+// Copyright (c) SimpleStaking and Tezedge Contributors
+// SPDX-License-Identifier: MIT
+#![forbid(unsafe_code)]
+
+// use crate::monitors::resource::Alerts;
+use crate::monitors::alerts::Alerts;
+use shiplift::Docker;
+use slog::{error, info, Logger};
+use std::path::PathBuf;
+use std::sync::atomic::{AtomicBool, Ordering};
+use std::sync::Arc;
+use tokio::task::JoinHandle;
+use tokio::time::{sleep, Duration};
+
+use crate::configuration::AlertThresholds;
+use crate::deploy_with_compose::{
+    cleanup_docker, restart_sandbox, restart_stack, stop_with_compose,
+};
+use crate::monitors::deploy::DeployMonitor;
+use crate::monitors::resource::{ResourceMonitor, ResourceUtilizationStorage};
+use crate::slack::SlackServer;
+
+pub mod alerts;
+pub mod deploy;
+pub mod resource;
+
+// TODO: get this info from docker (shiplift needs to implement docker volume inspect)
+// path to the volumes
+pub const TEZEDGE_VOLUME_PATH: &str =
+    "/var/lib/docker/volumes/deploy_monitoring_tezedge-shared-data/_data";
+pub const OCAML_VOLUME_PATH: &str =
+    "/var/lib/docker/volumes/deploy_monitoring_ocaml-shared-data/_data";
+
+pub fn start_deploy_monitoring(
+    compose_file_path: PathBuf,
+    slack: SlackServer,
+    interval: u64,
+    log: Logger,
+    running: Arc<AtomicBool>,
+    cleanup_data: bool,
+) -> JoinHandle<()> {
+    let docker = Docker::new();
+    let deploy_monitor =
+        DeployMonitor::new(compose_file_path, docker, slack, log.clone(), cleanup_data);
+    tokio::spawn(async move {
+        while running.load(Ordering::Acquire) {
+            if let Err(e) = deploy_monitor.monitor_stack().await {
+                error!(log, "Deploy monitoring error: {}", e);
+            }
+            sleep(Duration::from_secs(interval)).await;
+        }
+    })
+}
+
+pub fn start_sandbox_monitoring(
+    compose_file_path: PathBuf,
+    slack: SlackServer,
+    interval: u64,
+    log: Logger,
+    running: Arc<AtomicBool>,
+    cleanup_data: bool,
+) -> JoinHandle<()> {
+    let docker = Docker::new();
+    let deploy_monitor =
+        DeployMonitor::new(compose_file_path, docker, slack, log.clone(), cleanup_data);
+    tokio::spawn(async move {
+        while running.load(Ordering::Acquire) {
+            if let Err(e) = deploy_monitor.monitor_sandbox_launcher().await {
+                error!(log, "Sandbox launcher monitoring error: {}", e);
+            }
+            sleep(Duration::from_secs(interval)).await;
+        }
+    })
+}
+
+pub fn start_resource_monitoring(
+    interval: u64,
+    log: Logger,
+    running: Arc<AtomicBool>,
+    ocaml_resource_utilization: ResourceUtilizationStorage,
```

Comment:
@adonagy 
to be more flexible for future sandbox multinode support or single node run,
I would change this to: `resource_utilization: Vec<ResourceUtilizationStorage>`